### DISCUSSION TITLE
How to add libsodium to cmake

### DISCUSSION THREAD

**Discussion by briandowns** [Jason2605/Dictu#699] [General]
I'm really bad with cmake. Was curious if you (@Jason2605) could tell me how I can add libsodium to cmake. It's critical for some things I'm working on. ...pr inbound soon.

**Comment by Jason2605** [comment]
Good question, I'm currently struggling myself a little to get it included ☹️. Will keep having a play about and see if i can get it sorted

**Reply by Jason2605** [reply]
I'm not entirely sure why but on my feature/sodium branch, the CMake files in there seem to work and everything builds correctly if I include <sodium.h> into the CLI main.c, but any other place it can't find the header...

**Reply by briandowns** [reply]
It's weird that libsodium is such a pain. I even found a bunch of different cmake files that address exactly that. I'll hold off on doing much more libsodium work. I have a branch that I was going to add some key generation stuff, and work on some basic encryption. Sounds like you have some stuff in the works as well.

**Reply by Jason2605** [reply]
That branch was purely just trying to get libsodium working with CMake and to push it up if you wanted to see where I got with it. Nothing planned with the actual library from my end! It's such a pain and it probably helps I'm not hugely clued up on CMake to be honest. Will keep going to see if I can get it properly linked

Feel free to try any of those CMake files too

**Reply by briandowns** [reply]
Oh gotcha. Thank you so much! I appreciate that. :) 

I have a fork of the language from years ago where I removed cmake in favor of standard make. If that makes things any easier, I'm happy to bring in those changes. I don't have any strong opinions either way.

**Reply by Jason2605** [reply]
No worries!

We will want to keep CMake as it allows us to be a lot more cross-platform since we can generate build files for specific platforms. Just need to figure out the magic that it is 🤯

**Reply by briandowns** [reply]
It's some serious witchcraft.

**Reply by briandowns** [reply]
Ugh, libarchive is being a nightmare too.

**Comment by briandowns** [comment]
Now 3... Biggest pains ever.

* ncurses
* libarchive
* libsodium

**Comment by Jason2605** [comment]
Hah yeah I’ll have another try at getting those to work, no promises tho cmake is magic 😂

**Comment by briandowns** [comment]
Got ncurses to compile in. Should have a module on the way.
